ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
+        parameter_metadata=ENCODER_METADATA["TVMNASNetEncoder"]["model_variant"],
     )
 
-    embed_size: int = schema_utils.PositiveInteger(
-        default=512,
-        description="The patch embedding size, the output size of the mixer if avg_pool is true.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["embed_size"],
+
+@register_encoder_config("mnasnet_torch", IMAGE)
+@dataclass
+class TVMNASNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["mnasnet_torch"],
+        default="mnasnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    token_size: int = schema_utils.PositiveInteger(
-        default=2048,
-        description="The per-patch embedding size.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["token_size"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["0_5", "0_75", "1_0", "1_3"],
+        default="0_5",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVMNASNetEncoder"]["model_variant"],
     )
 
-    channel_dim: int = schema_utils.PositiveInteger(
-        default=256,
-        description="Number of channels in hidden layer.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["channel_dim"],
+
+@register_encoder_config("mobilenetv2_torch", IMAGE)
+@dataclass
+class TVMobileNetV2EncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["mobilenetv2_torch"],
+        default="mobilenetv2_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    num_layers: int = schema_utils.PositiveInteger(
-        default=8,
-        description="The depth of the network (the number of Mixer blocks).",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["num_layers"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["base"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVMobileNetV2Encoder"]["model_variant"],
     )
 
-    avg_pool: bool = schema_utils.Boolean(
-        default=True,
-        description="If true, pools output over patch dimension, outputs a vector of shape (embed_size). If false, "
-        "the output tensor is of shape (n_patches, embed_size), where n_patches is img_height x img_width "
-        "/ patch_size².",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["avg_pool"],
+
+@register_encoder_config("mobilenetv3_torch", IMAGE)
+@dataclass
+class TVMobileNetV3EncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["mobilenetv3_torch"],
+        default="mobilenetv3_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional